### PR TITLE
feat(Catalog Management): support change notices and plan id in offer…

### PR DESCRIPTION
**Pull request by benbuchanan**
…ing instance create

## PR summary

Need these changes to update the catalog management terraform provider.
- Now support plan id on offering instance create
- New field called change notices on an offering version

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

Just adding new fields that our api now supports.  Needed to update our terraform provider.

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: catalogmanagementv1/catalog_management_v1_examples_test.go [hunk 1]
```diff
@@ -377,6 +377,31 @@ var _ = Describe(`CatalogManagementV1 Examples Tests`, func() {
 			offeringID = *offering.ID
 		})
 
+		It(`GetOfferingChangeNotices request example`, func() {
```
Comment (padamstx) [inline catalogmanagementv1/catalog_management_v1_examples_test.go:380]: There are no examples for the get_plan, delete_plan, consumable_plan or set_deprecate_plan operations.

File: catalogmanagementv1/catalog_management_v1_integration_test.go [hunk 2]
```diff
@@ -610,6 +610,12 @@ var _ = Describe(`CatalogManagementV1 Integration Tests`, func() {
 				Evaluations: []catalogmanagementv1.Evaluation{*evaluationModel},
```
Comment (padamstx) [inline catalogmanagementv1/catalog_management_v1_integration_test.go:610]: No integration tests for the get_plan, delete_plan, consumable_plan, set_deprecate_plan operations.

**Review by padamstx — CHANGES_REQUESTED** [review]
1. Need to re-gen with the latest version of the SDK generator in order to leverage the new support for error handling in the Go SDK and Terraform providers.
2. Need to add working examples code and integration tests for the GetPlan, DeletePlan, ConsumablePlan, and SetDeprecatePlan operations.

**Comment by benbuchanan** [comment]
@padamstx Our API does not have the ability to create a plan unless the token making the call is an approver (Plans are created in Partner Center and they call us with an approver token to create them).  We just provide another way to manage them with the delete, deprecate, publish.

Is there precedence for running tests with a different token?

**Comment by benbuchanan** [comment]
example tests:
```
Ran 709 of 738 Specs in 48.630 seconds
SUCCESS! -- 709 Passed | 0 Failed | 0 Pending | 29 Skipped
PASS
ok  	github.com/IBM/platform-services-go-sdk/catalogmanagementv1	49.332s
```

integration tests:
```
Ran 720 of 776 Specs in 43.556 seconds
SUCCESS! -- 720 Passed | 0 Failed | 0 Pending | 56 Skipped
PASS
ok  	github.com/IBM/platform-services-go-sdk/catalogmanagementv1	44.380s
```

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**